### PR TITLE
WIP: Visual keybind representation.

### PR DESCRIPTION
**Pull request by marceliwac**
The patch aims to add a visual representation of the set keybinds (as outlined in #1929)

At the moment the implementation is just a rough draft of what the outcome should look like. Some help would be needed to put the CSS in correct place and possibly target some issues with different modifier keys (ctrl, alt etc.)

# Help needed

### CODE REVIEW

**Comment by BronzW** [comment]
What do you mean by putting css in the correct place?

**Comment by BronzW** [comment]
Inline styling works but using tailwind is better as it is what we use for kibbeh

**Comment by amitojsingh366** [comment]
ok ive said this before as-well but if you are thinking of making design change please confirm with @ajmnz beforehand, as if he/ben won't like it its gonna be rejected and all your time wasted.

File: kibbeh/src/modules/keyboard-shortcuts/ChatKeybind.tsx [hunk 1]
```diff
@@ -2,6 +2,7 @@ import React, { useEffect, useState } from "react";
 import { recordKeyCombination } from "react-hotkeys";
 import { useKeyMapStore } from "../../global-stores/useKeyMapStore";
 import { Button } from "../../ui/Button";
+import {VisualKeybind} from "./VisualKeybind";
```
Comment (dhenson02) [inline kibbeh/src/modules/keyboard-shortcuts/ChatKeybind.tsx:5]: try to keep formatting consistent with the rest of the code - `import { VisualKeybind } ...`

File: kibbeh/src/modules/keyboard-shortcuts/InviteKeybind.tsx [hunk 2]
```diff
@@ -2,6 +2,7 @@ import React, { useEffect, useState } from "react";
 import { recordKeyCombination } from "react-hotkeys";
 import { useKeyMapStore } from "../../global-stores/useKeyMapStore";
 import { Button } from "../../ui/Button";
+import {VisualKeybind} from "./VisualKeybind";
```
Comment (dhenson02) [inline kibbeh/src/modules/keyboard-shortcuts/InviteKeybind.tsx:5]: here too

**Review by dhenson02 — CHANGES_REQUESTED** [review]
- keep consistent formatting with the rest of the code
- use i18n for any user-facing static text in the UI

File: kibbeh/src/modules/keyboard-shortcuts/ChatKeybind.tsx [hunk 3]
```diff
@@ -39,7 +40,7 @@ export const ChatKeybind: React.FC<ChatKeybindProps> = ({ className }) => {
       <div className={`ml-4`}>
         toggle chat keybind:{" "}
         <span className={`font-bold text-lg`}>
-          {active ? "listening" : CHAT}
+          {active ? "listening" : <VisualKeybind keybind={CHAT.toString()} />}
```
Comment (dhenson02) [inline kibbeh/src/modules/keyboard-shortcuts/ChatKeybind.tsx:43]: why are all of these calling `.toString()`?  doesn't appear they will ever be returned as anything but a string already

**Review by dhenson02 — APPROVED** [review]

**Comment by marceliwac** [comment]
Thanks for the feedback @BronzW @amitojsingh366 @dhenson02!

This PR at the moment serves more of a feature idea that presents a possible outcome on what the changes could look like, rather than the how they should be implemented (hence the WIP tag). I'm happy to acknowledge that @ajmnz is responsbile for the design, so I'll happily take their lead on what the UI should look like. I thought I'd address some of the feedback you guys have so here goes:

- **CSS implementation**
I'm not very familiar with tailwind CSS so the current code aimed just to show what the keybind visuals would look like. I'll happily implement this correctly via tailwind once @ajmnz signs off on the redesign; for now I just used the inline styles to mock it up.

- **`toString()` conundrum**
Again, perhaps through fault of my own, lack of experience in TypeScript led me to this dirty fix to get rid of some of the compilation errors that arose from the `KeySequence` type, which declared few different types. I didn't take time to solve it, seeing as it's more of a mockup at this stage.
This also brings me to a second point, which is described in the [linked issue](https://github.com/benawad/dogehouse/issues/1929): _the displayed keys sometimes apply modifiers to the keys that are used (i.e. [Shift] + [a] will be displayed as Shift + A and not Shift + a (lowercase "a")_. 
Depending on how we collectively decide to tackle this aspect, it will either need changing to some other type, which can then be used for `keybind` prop of the `VisualKeybind` component, or we can just get rid of `toString()` in whichever way will satisfy the compiler.

- **i18n**
I'm not sure how the displayed keys will differ between different languages, but I don't suppose this PR will warrant any changes on that front. It appears that the current version of the code is not using any additional formatting for the keybinds and in general, key codes (i.e. Control -> Ctrl) tend to be the same across languages.

- **Key acronyms**
Somewhat linked to the above, one of the parts I'm unsure about is what the keys should be abbreviated to. For example, the **meta** key on the Mac is a **cmd** (or command) key and **alt** is an **option**, but this might differ on other systems like Windows or Linux. It would be nice to adapt to those different systems depending on the keyboard layout/os combination that is being used, but I would have to look into whether the [react-hotkeys](https://www.npmjs.com/package/react-hotkeys) provides support for that. If so, that shouldn't be too difficult, otherwise we can either scratch it completely and just use the keybinds returned by the **react-hotkeys** or write some logic that inferes these (which will lead to a bit of a rabbit hole in terms of deciding which systems should this be tested on, how to get the user OS etc.).

Thank you all again for the feedback. I'm looking forward to hearing from @ajmnz and moving forward with this.

File: kibbeh/src/modules/keyboard-shortcuts/ChatKeybind.tsx (hunk 3 above)
Comment (marceliwac) [reply kibbeh/src/modules/keyboard-shortcuts/ChatKeybind.tsx:43]: Please see my [comment](https://github.com/benawad/dogehouse/pull/1935#issuecomment-817143927) below.

**Comment by ajmnz** [comment]
This is no longer needed (#2282)

**Pull request closed without merging**